explorer/interpreter/type_checker.cpp

```diff
@@ -1543,16 +1594,18 @@ auto TypeChecker::Substitute(const Bindings& bindings,
       }
       ConstraintTypeBuilder builder(arena_,
                                     constraint.self_binding()->source_loc());
-      ErrorOr<Success> result =
-          builder.AddAndSubstitute(*this, &constraint, builder.GetSelfType(),
-                                   builder.GetSelfWitness(), bindings,
-                                   /*add_lookup_contexts=*/true);
+      // Diagnostics are discarded.
```

Comment:
According to line 1605, I don't think this is discarded, I think it should never fail

Reply:
Made more precise.